src/tiled.frame.hh

```diff
@@ -0,0 +1,61 @@
+#ifndef H_ACQUIRE_STORAGE_ZARR_TILED_FRAME_V0
+#define H_ACQUIRE_STORAGE_ZARR_TILED_FRAME_V0
+
+#include <cassert>
+#include <cmath>
+#include <functional>
+#include <iterator>
+#include <stdexcept>
+#include <unordered_map>
+#include <vector>
+
+#include "device/props/components.h"
+
+#include "prelude.h"
+
+namespace acquire::sink::zarr {
+struct TileShape
+{
+    struct tile_shape_dims_s
+    {
+        uint32_t width, height, planes;
+    } dims;
+};
+
+class TiledFrame
+{
+  public:
+    TiledFrame() = delete;
+    TiledFrame(const VideoFrame* frame,
+               const ImageShape&,
+               const TileShape& tile_shape);
+    TiledFrame(const TiledFrame&) = delete;
+    ~TiledFrame();
+
+    size_t bytes_of_image() const;
+
+    uint64_t frame_id() const;
+    uint8_t* data() const;
```

Comment:
`frame_id()` and `data()` should be made private. `bytes_of_image()` is used in a unit test but otherwise isn't used outside the class. Is there a way to make it private?

Reply:
`frame_id()` and `data()` don't actually appear to be used at all. I guess they're vestigial, so I'll get rid of em til they're needed. Guess I'll do the same with `bytes_of_image()`.